packages/nextjs/hooks/scaffold-eth/useAccountBalance.ts

```diff
@@ -0,0 +1,20 @@
+import { useEffect } from "react";
+import { useTargetNetwork } from "./useTargetNetwork";
+import { useQueryClient } from "@tanstack/react-query";
+import { UseBalanceParameters, useBalance, useBlockNumber } from "wagmi";
+
+type UseAccountBalanceParameters = UseBalanceParameters & { watch?: boolean };
+
+export const useAccountBalance = ({ watch = true, ...useBalanceParameters }: UseAccountBalanceParameters) => {
+  const { targetNetwork } = useTargetNetwork();
+  const queryClient = useQueryClient();
+  const { data: blockNumber } = useBlockNumber({ watch, chainId: targetNetwork.id });
+  const { queryKey, ...restUseBalanceReturn } = useBalance(useBalanceParameters);
+
+  useEffect(() => {
+    queryClient.invalidateQueries({ queryKey });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [blockNumber, queryClient]);
```

Comment:
We should remove `queryClient` from this dependency since it's an object this `useEffect` will be triggered unnecessarily a lot of time, That's why I didn't kept it as a dependency when we had this logic inlined in FaucetButton / Balance because it was triggering for very small user interactions

Reply:
it's an object but it's memoized, so it doesn't add additional rerenders. At least for me, probably you saw other cases. But removed it anyway

Reply:
> it's an object but it's memoized, so it doesn't add additional rerenders.

Ohh yeah yeah my bad, Actually I had `queryKey` as dependency to `useEffect` which was causing it to unnecessarily re-render that's why I removed it at https://github.com/scaffold-eth/scaffold-eth-2/pull/700/commits/c42a9e3b64ae640c76655914e6ad634094d0e242, but you already didn't had it as dependency so it was fine, thanks  🙌